### PR TITLE
Fix Teardown documentation

### PR DESCRIPTION
**Pull request by DomGarguilo**
Addresses concerns about the 2.x documentation from https://github.com/apache/accumulo/issues/3287

### CODE REVIEW

File: _docs-2/development/iterators.md [hunk 1]
```diff
@@ -408,12 +408,16 @@ On an instance of an Iterator: `init` is always called before `seek`, `seek` is
...
-As mentioned, instance of Iterators may be torn down inside of the server transparently. When a complex
-collection of iterators is performing some advanced functionality, they will not be torn down until a Key-Value
-pair is returned out of the "stack" of Iterators (and added into the batch of Key-Values to be returned
-to the caller). Being torn-down is equivalent to a new instance of the Iterator being creating and `deepCopy`
-being called on the new instance with the old instance provided as the argument to `deepCopy`. References
-to the old instance are removed and the object is lazily garbage collected by the JVM.
+As mentioned, instances of iterators may be torn down inside the server transparently. When a 
+complex collection of iterators is performing some advanced functionality, they will not be torn 
+down until a Key-Value pair is returned out of the "stack" of iterators (and added into the batch of
+Key-Values to be returned to the caller). Being torn down means that the iterator stack is recreated
```
Comment (ivakegg) [inline _docs-2/development/iterators.md:414]: ... to be returned to the caller), or the iterator is yielded.  Being torn down means ...

File: _docs-2/development/iterators.md [hunk 2]
```diff
@@ -408,12 +408,16 @@ On an instance of an Iterator: `init` is always called before `seek`, `seek` is
...
+as new instances with the original options, and then it is seeked from where it left off. References
+to the old instances are removed and the objects are lazily garbage collected by the JVM.
```
Comment (ivakegg) [inline _docs-2/development/iterators.md:416]: The sentence about the references to the old instances being garbace collected can be removed.

**Review by ivakegg — CHANGES_REQUESTED** [review]
see comments

File: _docs-2/development/iterators.md (hunk 2 above)
Comment (DomGarguilo) [reply _docs-2/development/iterators.md]: Addressed in fd55b58

File: _docs-2/development/iterators.md (hunk 1 above)
Comment (DomGarguilo) [reply _docs-2/development/iterators.md]: Addressed in fd55b58

**Comment by DomGarguilo** [comment]
Once these changes are finalized I'll make a PR to incorporate them into 1.10 as well

File: _docs-2/development/iterators.md [hunk 3]
```diff
@@ -408,12 +408,16 @@ On an instance of an Iterator: `init` is always called before `seek`, `seek` is
...
-to the old instance are removed and the object is lazily garbage collected by the JVM.
+Instances of iterators may be torn down inside the server transparently. When a complex collection
+of iterators is performing advanced functionality, they will not be torn down until a Key-Value pair
+is returned out of the "stack" of iterators (and added into the batch of Key-Values to be returned
+to the caller), or the iterator is yielded.
+
+When an iterator is torn down, the entire stack is dropped and no state is preserved. Only the last
+key returned (or the yielded position), original options, and seek range are retained. When the scan
+is continued, the iterator stack is rebuilt, re-initialized using the original options. The stack
+is then seeked with the original range, and the start key is replaced by the last key returned (or
```
Comment (Manno15) [inline _docs-2/development/iterators.md:419]:
```suggestion
key returned (or the yielded position), original options, and seek range are retained. When the scan
is continued, the iterator stack is rebuilt and re-initialized using the original options. The stack
is then seeked with the original range, and the start key is replaced by the last key returned (or
```

**Review by Manno15 — APPROVED** [review]
Just small grammar nits.

**Pull request merged**